File: frontend/src/types/report.ts
export interface BookingReport {
  booking_id: number;
  user_id: number;
  room_id: number;
  title: string;
  start_time: string;
  end_time: string;
  status: string;
}

export interface BookingReportSummary {
  total_bookings: number;
  confirmed_bookings: number;
  cancelled_bookings: number;
  completed_bookings: number;
  total_booked_hours: number;
}

File: backend/app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel


class BookingReportResponse(BaseModel):
    booking_id: int
    user_id: int
    room_id: int
    title: str
    start_time: datetime
    end_time: datetime
    status: str


class BookingReportSummaryResponse(BaseModel):
    total_bookings: int
    confirmed_bookings: int
    cancelled_bookings: int
    completed_bookings: int
    total_booked_hours: float